atomic_reactor/plugins/post_import_image.py

```diff
@@ -77,30 +77,48 @@ def run(self):
                                   build_json_dir=self.build_json_dir,
                                   namespace=metadata.get('namespace', None))
         osbs = OSBS(osbs_conf, osbs_conf)
-
+        imagestream = None
         try:
-            osbs.get_image_stream(self.imagestream)
+            imagestream = osbs.get_image_stream(self.imagestream)
         except OsbsResponseException:
             if self.insecure_registry is not None:
                 kwargs['insecure_registry'] = self.insecure_registry
 
             self.log.info("Creating ImageStream %s for %s", self.imagestream,
                           self.docker_image_repo)
 
-            # Tags are imported automatically on creation
-            osbs.create_image_stream(self.imagestream, self.docker_image_repo,
-                                     **kwargs)
-        else:
-            self.log.info("Importing new tags for %s", self.imagestream)
-
-            attempts = 0
-            while not osbs.import_image(self.imagestream):
-                attempts += 1
+            imagestream = osbs.create_image_stream(self.imagestream,
+                                                   self.docker_image_repo,
+                                                   **kwargs)
+        self.log.info("Importing new tags for %s", self.imagestream)
 
-                if attempts >= self.import_attempts:
-                    msg = "Failed to import new tags for %s"
-                    raise RuntimeError(msg % self.imagestream)
-
-                self.log.info("no new tags, will retry after %d seconds (%d/%d)",
-                              self.retry_delay, attempts, self.import_attempts)
-                sleep(self.retry_delay)
+        primaries = []
+        try:
+            primaries = self.workflow.build_result.annotations['repositories']['primary']
+        except (TypeError, KeyError):
+            pass
+
+        failures = False
+        for s in primaries:
+            tag_image_name = ImageName.parse(s)
+            tag = tag_image_name.tag
+            try:
+                osbs.ensure_image_stream_tag(imagestream.json(), tag)
+                self.log.info("Imported ImageStreamTag: (%s)", str(tag))
+            except OsbsResponseException:
+                failures = True
+                self.log.info("Could not import ImageStreamTag: (%s)", str(tag))
```

Comment:
Should the be `error` level instead of `info`?

*Same comment regarding unnecessary usage of `str(tag)`